modules/javafx.swing/src/main/java/com/sun/javafx/embed/swing/Disposer.java

```diff
@@ -89,6 +89,10 @@ public static WeakReference addRecord(Object target, DisposerRecord rec) {
         return ref;
     }
 
+    /**
+     * Remove the DisposerRecord object to prevent memory leak
+     * @param ref Weak reference of the object to be removed
+     */
```

Comment:
Looks good, however I think you could say this a bit more general, no need to mention that it is to prevent memory leaks -- it's just normal that you should remove the record if there is no need for that specific dispose task anymore because it was already disposed in another way.

So how about:

Unregisters a previously registered {@link DisposerRecord}, removing it from the list of records to dispose off when the original target goes out of scope. This is useful in cases where the original target is still in use, but the record was already disposed off by other means.